### ISSUE TITLE
constexpr tvec4<T, P>::tvec4

### ISSUE THREAD

**Issue by ma30002000** [g-truc/glm#556]
In GLM 0.9.8.1,

template 
GLM_FUNC_QUALIFIER GLM_CONSTEXPR_SIMD tvec4::tvec4(A a, B b, C c, D d)

should rather be declared

template 
GLM_FUNC_QUALIFIER GLM_CONSTEXPR tvec4::tvec4(A a, B b, C c, D d) :

since no SIMD implementation actually exists, allowing the ctor to be used in constexprs when GLM_ARCH != GLM_ARCH_PURE

**Comment by ma30002000** [comment]
template <typename A, typename B, typename C, typename D>
GLM_FUNC_QUALIFIER GLM_CONSTEXPR_SIMD tvec4<T, P>::tvec4(A a, B b, C c, D d) 

should rather be declared

template <typename A, typename B, typename C, typename D>
GLM_FUNC_QUALIFIER GLM_CONSTEXPR tvec4<T, P>::tvec4(A a, B b, C c, D d) :

since no SIMD implementation actually exists, allowing the ctor to be used in constexprs when GLM_ARCH != GLM_ARCH_PURE

**Comment by patrikhuber** [comment]
I'm getting a very similar error with glm-0.9.8.1 as well as the latest master branch, however, only when `GLM_FORCE_UNRESTRICTED_GENTYPE` is defined:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	C2475	'glm::tvec4<float,glm::aligned_lowp>::tvec4': redefinition; 'constexpr' specifier mismatch	glm-0.9.8.1\glm\detail\type_vec4_simd.inl	344	
Error	C2475	'glm::tvec4<float,glm::aligned_mediump>::tvec4': redefinition; 'constexpr' specifier mismatch	glm-0.9.8.1\glm\detail\type_vec4_simd.inl	351	
Error	C2475	'glm::tvec4<float,3>::tvec4': redefinition; 'constexpr' specifier mismatch	glm-0.9.8.1\glm\detail\type_vec4_simd.inl	358	
```

These are the lines:
`GLM_FUNC_QUALIFIER GLM_CONSTEXPR_CTOR tvec4<float, aligned_lowp>::tvec4()`
`GLM_FUNC_QUALIFIER GLM_CONSTEXPR_CTOR tvec4<float, aligned_mediump>::tvec4()`
`GLM_FUNC_QUALIFIER GLM_CONSTEXPR_CTOR tvec4<float, aligned_highp>::tvec4()`

I wonder why it's parsing `type_vec4_simd.inl` at all, I'm using the default flags/defines, no SSE or anything.

**Comment by patrikhuber** [comment]
Possibly related: #555. I'm on VS2015U3 as well.

**Comment by Groovounet** [comment]
This issue should be fixed in GLM 0.9.8 and master branches.

Thanks for reporting,
Christophe
